### ISSUE TITLE
RQG simple-aggregates workload fails

### ISSUE THREAD

**Issue by nrainer-materialize** [MaterializeInc/materialize#25926]
### What version of Materialize are you using?

ec9d3fc6e5adf8f7f9851855f013bc11057c3c8e

### What is the issue?

https://buildkite.com/materialize/nightlies/builds/6844#018e2d4a-0d77-4974-8816-5e86d8f77f82

Also observed in:
* https://buildkite.com/materialize/nightlies/builds/6841#018e2cae-ca46-4bfa-b371-61d577f08cd2
* https://buildkite.com/materialize/nightlies/builds/6845#018e2d70-6fdd-4fb9-8379-25025c16cb2e

**Comment by nrainer-materialize** [comment]
I will disable the test for now and try to bisect the issue.

CC @aalexandrov

**Comment by nrainer-materialize** [comment]
This [broke](https://buildkite.com/materialize/nightlies/builds/6849) with https://github.com/MaterializeInc/materialize/pull/25914, which enabled `enable_explain_broken` in CI.

@aalexandrov: Can we make RQG work with this?

CC @ParkMyCar

**Comment by aalexandrov** [comment]
I'm not even sure how this will contribute to the issue—the feature flag just fences the `EXPLAIN BROKEN` sytnax.

If RQG doesn't generate `EXPLAIN BROKEN` queries we will never hit the `if broken` branch here

https://github.com/MaterializeInc/materialize/blob/5c7eeb50cf452e5636edc1762b7d770a7a2d0460/src/adapter/src/explain/optimizer_trace.rs#L60-L96

so at runtime there will be no difference between runs with the flags `on` and `off`.

@nrainer-materialize: what was the reason for turning the default `on`?

**Comment by aalexandrov** [comment]
This seems like a correctness bug as this query returns a different result in Postgres:

```sql
SELECT
    (a2.f2 + NULLIF (a1.f1, a1.f1)) AS c1,
    (a2.f2 + NULLIF (a2.f1, a2.f2)) AS c2,
    (a2.f2) AS c3,
    (count(DISTINCT NULLIF (a2.f2, a2.f1))) AS agg1,
    (min(DISTINCT a2.f1)) AS agg2
FROM t1 AS a1 RIGHT JOIN t2 AS a2 USING(f2, f1)
WHERE
    NOT (a1.f2 IS NOT NULL)
        AND
    NULLIF (a2.f1, a1.f2) = NULLIF (a2.f2, a1.f1)
GROUP BY 1, 2, 3;
```

I'm truing to see if this is a regression introduced in one of our recent versions, though.

**Comment by nrainer-materialize** [comment]
> so at runtime there will be no difference between runs with the flags on and off.

My understanding is that the RQG build fails if this flag is enabled. Bisecting the RQG failure pointed to this commit. Though, let me know if this absolutely does not make sense and I will re-conduct the bisecting.

> @nrainer-materialize: what was the reason for turning the default on?
We strive to enable all feature flags in CI when possible. Most feature flags temporarily keep some functionality behind the fence for some time and will eventually be removed when the functionality reaches the public phase. @ParkMyCar, what does the envisioned lifecycle for `enable_explain_broken` look like?

**Comment by aalexandrov** [comment]
> @ParkMyCar, what does the envisioned lifecycle for enable_explain_broken look like?

The flag was added by me because at some point in the past `EXPLAIN BROKEN` was known to leak memory, but at the same time it was useful for debugging. The idea was that whoever wants to use it needs to enable a feature flag in order to protect random workloads causing OOMs.

**Comment by aalexandrov** [comment]
> Bisecting the RQG failure pointed to this commit. Though, let me know if this absolutely does not make sense and I will re-conduct the bisecting.

Yeah I am running the query reported by the RQG failure on other versions of materialize (up to `v0.70.2`) and on all of them the result is empty, while on Postgres the result is non-empty.

**Comment by nrainer-materialize** [comment]
Ok, we can disable this flag in that case for RQG then, I guess.

**Comment by aalexandrov** [comment]
Actually I'm starting to suspect that this is actually a bug in ... Postgres.

**Comment by aalexandrov** [comment]
The problem is not present in the latest Postgres version (16.2).

**Comment by aalexandrov** [comment]
@MaterializeInc/testing we need to either fugure out a way to detect and ignore the faulty query or we need to upgrade to a newer version of Postgres for our RQG tests.

**Comment by aalexandrov** [comment]
I think this should be fixed by #25964.
